src/pick.test.ts

```diff
@@ -75,4 +75,18 @@ describe('typing', () => {
       >();
     });
   });
+
+  test('multiple keys', () => {
```

Comment:
This test doesn't test what we want it to because it succeeds without any modifications to the implementation.

The thing is that the bug isn't even exactly a Remeda issue, it's more likely a Typescript issue - You can see that Typescript does eventually pick (no pun intended) the correct overload of the function when you finish typing; it would continue to type-check that the prop names given are actual keys of the data object (you can test this by adding this test to the unmodified version of `pick` and changing any of the props to an invalid name, or adding one in addition).

As far as I can tell, the only bug here seems related specifically to auto-completion...

Reply:
Thanks! I agree that it's certainly not a bug with Remeda but rather Typescript.